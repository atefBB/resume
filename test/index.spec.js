// import faker from "faker"
import { testHelper, mountHelper } from "./helpers"
import Index from "~/pages/index.vue"
jest.mock(`~/assets/yaml/about.yml`)
jest.mock(`~/assets/yaml/educations.yml`)
jest.mock(`~/assets/yaml/experiences.yml`)
jest.mock(`~/assets/yaml/header.yml`)
jest.mock(`~/assets/yaml/hobbies.yml`)
jest.mock(`~/assets/yaml/projects.yml`)

describe("Index", () => {
  testHelper(Index)

  it("print the page", () => {
    const factory = mountHelper(Index)
    const wrapper = factory()
    global.print = jest.fn()
    const btn = wrapper.find("div.corner-ribbon.top-right")
    btn.trigger("click")
    expect(global.print).toBeCalled()
  })
})
