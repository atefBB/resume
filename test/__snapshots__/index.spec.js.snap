// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Default Template 1`] = `
<section class="el-container is-vertical">
  <main class="el-main main-container" id="main">
    <div class="container">
      <div class="el-row">
        <div class="el-col el-col-12">
          <div class="header-left">
            <h1 id="name">
              David Turbert
            </h1>
            <h2 id="title">
              Senior Web or Lead Developer / Junior Data Scientist
            </h2>
          </div>
        </div>
        <div class="el-col el-col-12">
          <div>
            <ul class="icons">
              <li id="icons-github"><a target="_blank" href="https://www.github.com/dcyou"><i title="Github link" class="fab fa-github"></i></a></li>
              <li id="icons-linkedin"><a target="_blank" href="https://www.linkedin.com/in/dcyou"><i title="Linkedin link" class="fab fa-linkedin"></i></a></li>
              <li id="icons-twitter"><a target="_blank" href="https://twitter.com/david_turbert"><i title="Twitter link" class="fab fa-twitter"></i></a></li>
              <li id="icons-email"><a target="_blank" href="mailto:david@dcyou.fr"><i title="Email link" class="fas fa-at"></i></a></li>
              <li id="icons-phone"><a target="_blank" href="tel:+33.633.578.939"><i title="Phone link" class="fas fa-phone-alt"></i></a></li>
              <!---->
            </ul>
          </div>
        </div>
      </div>
    </div>
    <div class="container">
      <h3 id="about-title">
        About Me
      </h3>
      <div class="el-row">
        <div class="el-col el-col-8">
          <p><span class="el-avatar el-avatar--circle" id="about-profile-image"><img src="me.jpg" style="object-fit: fill;"></span></p>
        </div>
        <div class="el-col el-col-16">
          <p id="about-content">Hi, my name’s David. I'm Web Developer. My experience has allowed me to acquire solid knowledge in development, but not only. Whatever programming language is used, my knowledge of software engineering allows me to put into practice, with rigor, the best development practices. Curious by nature and self-taught I like to follow the new technologies news. Continuous learning is an indispensable thing for me, both professionally and personally.
          </p>
        </div>
      </div>
    </div>
    <div class="container">
      <h3 id="project-title">
        Projects
      </h3>
      <div class="el-card is-always-shadow" data="[object Object]" index="0">
        <div class="el-card__header">
          <div style="text-align: center;"><a href="https://github.com/dcyou/resume" class="el-link el-link--default is-underline">
              <!----><span class="el-link--inner"><i class="fab fa-github"></i>
        Resume
      </span>
              <!----></a></div>
        </div>
        <div class="el-card__body">
          <div>I started this project as a way of learning <mark>NuxtJS</mark> and <mark>Element</mark> component library. I have learned many skills through this and been I’m very proud of having this in my portfolio. If you don’t have a project as awesome as this I would advise you make one. I also explore the basis of <mark>Travis</mark> to compile this project and publish it on the <mark>Github page</mark> Finally, I am not a Web Designer but I would like to understand well the <mark>Dark Theme</mark> provide by Google. So this is why I manually modify the <mark>CSS</mark> to follow as many as possible <a href="https://material.io/design/color/dark-theme.html">Google recommandations</a>
          </div>
        </div>
      </div>
    </div>
    <div class="container">
      <h3 id="experience-title">
        Work experience
      </h3>
      <div class="el-card is-always-shadow" data="[object Object]" index="0">
        <div class="el-card__header">
          <div style="text-align: center;">
            <p><a href="https://david.dcyou.fr" class="el-link el-link--default is-underline">
                <!----><span class="el-link--inner"><i class="far fa-building"></i>
          DCyou
        </span>
                <!----></a>
              <div class="el-divider el-divider--vertical">
                <!---->
              </div> <i class="fas fa-map-marker-alt"></i> Annecy
            </p>
            <p>2018 - 2019</p>
            <p>founder</p>
          </div>
        </div>
        <div class="el-card__body">
          <div>
            <ul>
              <li>Website, Channel manager, Marketing (Google Ads etc.) and manual labor for <a href="https://chalet-charmelie.com">Chalet le Charmélie 1600</a> at Flaine</li>
              <li>
                Self-taught:
                <ul>
                  <li>
                    Jenkins: <a href="https://trainingcourses.page.link/jenkins">1</a>
                    <a href="https://trainingcourses.page.link/jenkins-2">2</a>
                  </li>
                  <li>
                    Python: <a href="https://trainingcourses.page.link/python-1">1</a>
                    <a href="https://trainingcourses.page.link/python-2">2</a>
                  </li>
                  <li>
                    <a href="https://trainingcourses.page.link/magento1">Magento 1</a>
                  </li>
                  <li>
                    <a href="https://trainingcourses.page.link/react">React / Firebase</a>
                  </li>
                  <li>
                    <a href="https://trainingcourses.page.link/react-native">React Native</a>
                  </li>
                  <li>
                    <a href="https://trainingcourses.page.link/silex">Silex</a>
                  </li>
                  <li>
                    <a href="https://trainingcourses.page.link/behat">BDD (Behat)</a>
                  </li>
                  <li>
                    <a href="https://trainingcourses.page.link/aws">AWS (EC2, RDS, Beanstalk, S3)</a>
                  </li>
                  <li>
                    Laravel 5.8 + VueJS 2.x
                  </li>
                </ul>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div class="el-card is-always-shadow" data="[object Object]" index="1">
        <div class="el-card__header">
          <div style="text-align: center;">
            <p><a href="https://www.firmenich.com" class="el-link el-link--default is-underline">
                <!----><span class="el-link--inner"><i class="far fa-building"></i>
          Firmenich
        </span>
                <!----></a>
              <div class="el-divider el-divider--vertical">
                <!---->
              </div> <i class="fas fa-map-marker-alt"></i> Geneva
            </p>
            <p>2014 - 2018</p>
            <p>Lead developer / DevOps / Junior data scientist</p>
          </div>
        </div>
        <div class="el-card__body">
          <div>
            <ul>
              <li>
                Lead developer
                <ul>
                  <li>By combine different technology, web applications for proof of concept</li>
                  <li>Elasticsearch with Meteor to build an intelligent search engine tailored</li>
                  <li>PHP with MySQL for an global application of user management and application management</li>
                  <li>NodeJS, create some packages for the applications (authentication with JWT, databases connections etc.) usage of some Design Pattern</li>
                  <li>NodeJS, Matlab, MySQL and MongoDB to build a complex app to improve business performance</li>

                  <li>Code review</li>
                  <li>Team training for Git best practices and development best practices</li>
                  <p>Languages: <mark>NodeJS</mark> <mark>React</mark> <mark>PHP</mark> <mark>Matlab</mark> </p>
                  <p>Databases: <mark>MySQL</mark> <mark>Elasticsearch</mark> <mark>MongoDB</mark> <mark>Oracle</mark> <mark>GraphQL</mark></p>
                </ul>
              </li>
              <li>
                DevOps
                <ul>
                  <li>From scratch, creation and support of a software factory. Environments of development, qualification and production for Matlab models (Windows moved to Linux Redhat) and another one for web applications (implementation from scratch Linux Redhat)</li>
                  <li>Management of Ubuntu server with Docker. Creation of image and scripting to simplify the usage of containers</li>
                  <p>Skills: <mark>Docker</mark> <mark>GIT</mark> <mark>Gitlab</mark> <mark>Gitlab CI</mark> <mark>Capistrano</mark> <mark>LAMP</mark> <mark>NGINX</mark> <mark>Matlab Production Server</mark></p>
                </ul>
              </li>
              <li>
                Junior data scientist
                <ul>
                  <li>Scripting SQL and PL/SQL to aggregate, clean and calculate data.</li>
                  <li>Analyze, organize and structured data to help principal scientists</li>
                  <li>Implementation of Matlab Unit testing</li>
                  <li>Official training by Mathworks:</li>
                  <ul>
                    <li>MATLAB Fundamentals</li>
                    <li>MATLAB Programming Techniques</li>
                    <li>Building Interactive Applications in MATLAB</li>
                    <li>Machine Learning with MATLAB</li>
                  </ul>
                </ul>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div class="el-card is-always-shadow" data="[object Object]" index="2">
        <div class="el-card__header">
          <div style="text-align: center;">
            <p><a href="https://www.greenweez.com/" class="el-link el-link--default is-underline">
                <!----><span class="el-link--inner"><i class="far fa-building"></i>
          Greenweez.com
        </span>
                <!----></a>
              <div class="el-divider el-divider--vertical">
                <!---->
              </div> <i class="fas fa-map-marker-alt"></i> Annecy
            </p>
            <p>2012 - 2014</p>
            <p>Web developer</p>
          </div>
        </div>
        <div class="el-card__body">
          <div>
            <ul>
              <li>Website development and maintenance</li>
              <li>Project management (mobile website, implementation of new payment and delivery method)</li>
              <li>Development of new order check out by using PHP object</li>
              <li>Implementation of a new search engine with Elasticsearch</li>
              <li>Internationalization to open the website to new countries (Spain, UK, Italy etc.)</li>
              <li>Code review to fix some security breach</li>
              <p>Skills: <mark>PHP</mark> <mark>JavaScript</mark> <mark>MySQL</mark> <mark>Elasticsearch</mark></p>
            </ul>
          </div>
        </div>
      </div>
      <div class="el-card is-always-shadow" data="[object Object]" index="3">
        <div class="el-card__header">
          <div style="text-align: center;">
            <p><a href="https://www.dipole.eu/" class="el-link el-link--default is-underline">
                <!----><span class="el-link--inner"><i class="far fa-building"></i>
          Dipole
        </span>
                <!----></a>
              <div class="el-divider el-divider--vertical">
                <!---->
              </div> <i class="fas fa-map-marker-alt"></i> Lyon
            </p>
            <p>2010 - 2012</p>
            <p>Student</p>
          </div>
        </div>
        <div class="el-card__body">
          <div>
            <ul>
              <li>
                Migration of a Workflow product (management of quality) from C++ to a web solution
                <p>Skills: <mark>PHP (Zend Framework)</mark> <mark>JavaScript (ExtJS)</mark> <mark>JGraph</mark> <mark>FusionChart</mark> <mark>PowerChart</mark></p>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div class="el-card is-always-shadow" data="[object Object]" index="4">
        <div class="el-card__header">
          <div style="text-align: center;">
            <p><a href="https://www.ret.fr/" class="el-link el-link--default is-underline">
                <!----><span class="el-link--inner"><i class="far fa-building"></i>
          ESS / RET
        </span>
                <!----></a>
              <div class="el-divider el-divider--vertical">
                <!---->
              </div> <i class="fas fa-map-marker-alt"></i> Annecy
            </p>
            <p>2008 - 2010</p>
            <p>Student</p>
          </div>
        </div>
        <div class="el-card__body">
          <div>
            <ul>
              <li>
                Work as an system and network administrator
                <p>Skills: <mark>OCS</mark> <mark>GLPI</mark> <mark>WIKI</mark> <mark>WSUS</mark> <mark>Windows servers administration</mark> <mark>hiring process</mark></p>
              </li>
              <li>
                Applications development for files treatment and accounting management
                <p>Skills: <mark>C#</mark> <mark>Crystal report</mark></p>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
    <div class="container">
      <h3 id="education-title">
        Education
      </h3>
      <ul class="el-timeline">
        <li class="el-timeline-item">
          <div class="el-timeline-item__tail"></div>
          <div class="el-timeline-item__node el-timeline-item__node--normal el-timeline-item__node--">
            <!---->
          </div>
          <!---->
          <div class="el-timeline-item__wrapper">
            <div class="el-timeline-item__timestamp is-top">
              2010 - 2012
            </div>
            <div class="el-timeline-item__content">
              <div class="el-card is-always-shadow">
                <!---->
                <div class="el-card__body">
                  <h4>
                    EPSI
                    <div class="el-divider el-divider--vertical">
                      <!---->
                    </div> <span><i class="fas fa-map-marker-alt"></i>
            Lyon
          </span></h4>
                  <p>Master’s degree</p>
                </div>
              </div>
            </div>
            <!---->
          </div>
        </li>
        <li class="el-timeline-item">
          <div class="el-timeline-item__tail"></div>
          <div class="el-timeline-item__node el-timeline-item__node--normal el-timeline-item__node--">
            <!---->
          </div>
          <!---->
          <div class="el-timeline-item__wrapper">
            <div class="el-timeline-item__timestamp is-top">
              2009 - 2010
            </div>
            <div class="el-timeline-item__content">
              <div class="el-card is-always-shadow">
                <!---->
                <div class="el-card__body">
                  <h4>
                    Partner Formation
                    <div class="el-divider el-divider--vertical">
                      <!---->
                    </div> <span><i class="fas fa-map-marker-alt"></i>
            Lyon
          </span></h4>
                  <p>Bachelor’s degree « IT Project Manager »</p>
                </div>
              </div>
            </div>
            <!---->
          </div>
        </li>
        <li class="el-timeline-item">
          <div class="el-timeline-item__tail"></div>
          <div class="el-timeline-item__node el-timeline-item__node--normal el-timeline-item__node--">
            <!---->
          </div>
          <!---->
          <div class="el-timeline-item__wrapper">
            <div class="el-timeline-item__timestamp is-top">
              2007 - 2009
            </div>
            <div class="el-timeline-item__content">
              <div class="el-card is-always-shadow">
                <!---->
                <div class="el-card__body">
                  <h4>
                    Maestris
                    <div class="el-divider el-divider--vertical">
                      <!---->
                    </div> <span><i class="fas fa-map-marker-alt"></i>
            Valence
          </span></h4>
                  <p>HND Application developer</p>
                </div>
              </div>
            </div>
            <!---->
          </div>
        </li>
      </ul>
    </div>
    <div class="container">
      <h3 id="hobbies-title">
        Activities and interests
      </h3>
      <div class="el-card is-always-shadow" id="hobbies-content">
        <!---->
        <div class="el-card__body">
          Sports (cycling, running, triathlon etc.), High Tech actualities (IOT, CTF etc.), epicurean, earth and environment, ‘Mens sana in corpore sano’

        </div>
      </div>
    </div>
  </main>
  <div class="corner-ribbon top-right sticky no-print">
    Print me
  </div>
  <footer class="el-footer" style="height: 60px;">
    <gh-btns-star slug="dcyou/resume"></gh-btns-star>
  </footer>
</section>
`;
